web/packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -523,19 +529,38 @@ export default class Codec {
     return buffer;
   }
 
+  static keyboardButtonMessageLength =
+    1 /* MessageType */ + 4 /* scanCode */ + 1 /* state */;
+
   // encodeKeyboardInput encodes a keyboard action.
   // Returns null if an unsupported code is passed.
   // | message type (5) | key_code uint32 | state byte |
   encodeKeyboardInput(code: string, state: ButtonState): Message | null {
-    const scanCode = this._keyScancodes[code];
-    if (!scanCode) {
+    const scanCodeOrCodes = this._keyScancodes[code];
+    if (!scanCodeOrCodes) {
+      console.warn(`unsupported key code: ${code}`);
       return null;
     }
-    const buffer = new ArrayBuffer(6);
+
+    // Normalize scanCodeOrCodes to always be an array
+    const scanCodes = Array.isArray(scanCodeOrCodes)
+      ? scanCodeOrCodes
+      : [scanCodeOrCodes];
+    const totalScanCodes = scanCodes.length;
+
+    // Calculate the total buffer size needed
+    const bufferSize = totalScanCodes * Codec.keyboardButtonMessageLength;
+    const buffer = new ArrayBuffer(bufferSize);
     const view = new DataView(buffer);
-    view.setUint8(0, MessageType.KEYBOARD_BUTTON);
-    view.setUint32(1, scanCode);
-    view.setUint8(5, state);
+
+    let offset = 0;
+    scanCodes.forEach(scanCode => {
```

Comment:
Now I see what you are doing here. You are actually sending multiple TDP messages in a single websocket write. We've never done that before, it's always been 1:1. 

I'm not sure there's a compelling enough reason to break from tradition here.

Reply:
Hmm good point, we do have some logic on the backend that may rely on this 1:1. I'll fix that.